portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -24,9 +34,79 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
-  @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("Hello Laura!");
+  private List<String> messages;
+
+    @Override
+    public void init() {
+      messages = new ArrayList<String>();
+    }
+
+    @Override 
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+      // Converting Array List to JSON.
+      String jsonVersion = convertToJson(messages);
+
+      Query query = new Query("Comments").addSort("comment", SortDirection.ASCENDING);
+      DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+      PreparedQuery results = datastore.prepare(query);
+
+      for (Entity entity : results.asIterable()) {
+        String comment = (String) entity.getProperty("comment");
+
+        Task task = new Task(comment);
+        jsonVersion.add(task);
+    }
+
+      // Send JSON string.
+      response.setContentType("application/json;");
+      response.getWriter().println(jsonVersion);
+    }
+
+    private String convertToJson(List messages) {
+      Gson gson = new Gson();
+      return gson.toJson(messages);
+    }
+
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+      // Get the input from the form.
+      String text = getParameter(request, "word-input", "");
+      boolean positive = Boolean.parseBoolean(getParameter(request, "positive", "false"));
```

Comment:
Suggestion: Since these look like they're all mutually exclusive it would be better to have a single enum rather than multiple booleans. Oh wait, just scrolled down and it looks like they aren't mutually exclusive, nvm then haha.